src/main/scala/Joint.scala

```diff
@@ -62,9 +131,13 @@ case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Doub
     * vector for the plane, and the second item is the distance of the plane from the origin.
     */
   def globalCoordinates: Seq[((Double, Double, Double), Double)] = {
-    val Nplane = DenseVector[Double](a, b, c)
-    val strike = (dipDirection - math.Pi / 2) % (2 * math.Pi) // CHECKED: strike = dipDirection - pi/2 (US convention)
-    val Nstrike = DenseVector[Double](math.cos(strike), math.sin(strike), 0.0)
+    var normalDirection = -1.0 // Negative needed simply because of how vertical planes are dealt with
```

Comment:
I think you can avoid using a `var` here and stick with the preferred `val` with something like the following. Scala's `if` statements produce a value.
```scala
val normalDirectrion = if (math.abs(c) < Joint.EPSILON) -1.0 else -c/math.abs(c)
```

Actually, I think this is kind of confusing anyways. I had to plot `-x/math.abs(x)` to understand what was going on. Maybe something like this would be better:
```scala
val normalDirection = if (c > -Joint.EPSILON) {
    -1.0
} else {
    1.0
}
```